src/mlpack/methods/gmm/CMakeLists.txt

```diff
@@ -24,23 +24,26 @@ endforeach()
 # the parent scope).
 set(MLPACK_SRCS ${MLPACK_SRCS} ${DIR_SRCS} PARENT_SCOPE)
 
+add_category(gmm_train "clustering")
 add_cli_executable(gmm_train)
 add_python_binding(gmm_train)
 add_julia_binding(gmm_train)
 add_go_binding(gmm_train)
 add_r_binding(gmm_train)
-add_markdown_docs(gmm_train "cli;python;julia;go;r" "clustering")
+add_markdown_docs(gmm_train "cli;python;julia;go;r" "")
 
+add_category(gmm_generate "clustering")
 add_cli_executable(gmm_generate)
 add_python_binding(gmm_generate)
 add_julia_binding(gmm_generate)
 add_go_binding(gmm_generate)
 add_r_binding(gmm_generate)
-add_markdown_docs(gmm_generate "cli;python;julia;go;r" "clustering")
+add_markdown_docs(gmm_generate "cli;python;julia;go;r" "")
 
+add_category(gmm_probability "clustering")
 add_cli_executable(gmm_probability)
 add_python_binding(gmm_probability)
 add_julia_binding(gmm_probability)
 add_go_binding(gmm_probability)
 add_r_binding(gmm_probability)
-add_markdown_docs(gmm_probability "cli;python;julia;go;r" "clustering")
+add_markdown_docs(gmm_probability "cli;python;julia;go;r" "")
```

Comment:
Since these programs are already split, do you think we could make a wrapper for them as they are?  Or would it need more modification?  (We can do that in a different PR of course, I am just curious. :))

Reply:
Some changes would be required in the `BINDING_*` macros, other than that there should not be any changes, but if we combine these now then documentation for other languages might get skewed because it does not already have a single file that can be used for rest of the languages. To tackle this, we will have to copy each `_main.cpp` file and create a new file that we will use for python.

Reply:
Ahh, good point.  Let's handle that in another PR in the future. :)